File: ios/VectorialData/VectorialData/Core/Config/AppConfig.swift
import Foundation

enum AppConfig {
    /// Canonical production base URL for the Vectorial Data API.
    /// Local development can override via VD_API_BASE_URL in scheme env vars.
    static let apiBaseURL: URL = {
        if let override = ProcessInfo.processInfo.environment["VD_API_BASE_URL"],
           let url = URL(string: override) {
            return url
        }
        return URL(string: "https://www.vectorialdata.com")!
    }()

    /// Deep-link scheme registered in Info.plist CFBundleURLTypes.
    static let urlScheme = "vectorialdata"

    /// Keychain service identifier for the auth token.
    static let keychainService = "com.vectorialdata.app.auth"
}

File: android/app/src/main/java/com/vectorialdata/app/core/config/AppConfig.kt
package com.vectorialdata.app.core.config

import com.vectorialdata.app.BuildConfig

/**
 * Mirror of iOS `AppConfig`. Canonical production base URL for the
 * Vectorial Data API; overridable per build type via BuildConfig.
 */
object AppConfig {
    /** Base URL for the Vectorial Data API (no trailing slash). */
    val apiBaseUrl: String = BuildConfig.API_BASE_URL.trimEnd('/')

    /** Deep-link scheme registered in the manifest. */
    const val URL_SCHEME = "vectorialdata"

    /** Client identifier sent to the backend so it emits the right magic link. */
    const val CLIENT = "android"
}
